### PR TITLE
Implement group membership CRUD and unit/acceptance testing

### PR DESCRIPTION
**Pull request by nmiodice**
## All Submissions:
-------------------------------------
* [YES] Have you added an explanation of what your changes do and why you'd like us to include them?
* [YES] I have updated the documentation accordingly.
* [YES] I have added tests to cover my changes.
* [YES] All new and existing tests passed.
* [YES] My code follows the code style of this project.
* [YES] I ran lint checks locally prior to submission.
* [YES] Have you checked to ensure there aren't other open PRs for the same update/change?

## What is the current behavior?
-------------------------------------
No resource for `azuredevops_group_membership` has been implemented so groups cannot be managed via AzDO

Issue Number: https://github.com/microsoft/terraform-provider-azuredevops/issues/127

## What is the new behavior?
-------------------------------------
- Support group membership CRUD

## Does this introduce a breaking change?
-------------------------------------
- [NO]

### CODE REVIEW

File: azuredevops/resource_group_membership_test.go [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+			{
+				// add the group membership
+				Config: tfStanzaWithMembership,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(tfNode, "id"),
+					resource.TestCheckResourceAttrSet(tfNode, "group"),
+					// this attribute specifies the number of members in the resource state. the
+					// syntax is how terraform maps complex types into a flattened map.
+					resource.TestCheckResourceAttr(tfNode, "members.#", "1"),
+					// testAccVerifyGroupMembershipMatchesState(),
```
Comment (nmiodice) [inline azuredevops/resource_group_membership_test.go]:
```suggestion
					testAccVerifyGroupMembershipMatchesState(),
```

File: azuredevops/resource_group_membership_test.go [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+				),
+			}, {
+				// remove the group membership
+				Config: tfStanzaWithoutMembership,
+				// Check: testAccVerifyGroupMembershipMatchesState(),
```
Comment (nmiodice) [inline azuredevops/resource_group_membership_test.go]:
```suggestion
				Check: testAccVerifyGroupMembershipMatchesState(),
```

File: azuredevops/resource_group_membership_test.go [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+		_, expectingMembership := s.RootModule().Resources["azuredevops_group_membership.membership"]
+
+		// The sleep here is to take into account some propegation delay that can happen with Group Membership APIs.
+		// If we want to go inspect the behavior of the service after a Terraform Apply, we'll need to wait a little bit
+		// before making the API call.
+		//
+		// Note: some thought was put behind keeping the time.sleep here vs in the provider implementation. After consideration,
+		// I decided to keep it here. Moving to the provider would (1) provide no functional benefit to the end user, (2) increase
+		// complexity and (3) be inconsistent with the UI and CLI behavior for the same operation.
+		time.Sleep(5 * time.Second)
```
Comment (TsuyoshiUshio) [inline azuredevops/resource_group_membership_test.go]: Is this required?

File: azuredevops/resource_group_membership_test.go [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+)
+
+/**
+ * Begin unit tests
+ */
+
+func TestGroupMembership_ComputeMembershipDiff_ResolvesDiffProperly(t *testing.T) {
+	// If you are curious about the use of map here, have a read through this article:
+	//	https://stackoverflow.com/questions/34018908/golang-why-dont-we-have-a-set-datastructure
+	type membershipsTestMeta struct {
```
Comment (TsuyoshiUshio) [inline azuredevops/resource_group_membership_test.go]: cool!

File: azuredevops/resource_group_membership_test.go (hunk 3 above)
Comment (nmiodice) [reply azuredevops/resource_group_membership_test.go]: Unfortunately the graph membership API, despite being a synchronous API, does not always reflect PUT/DELETE operations right away. So this is necessary for our testing to be consistent.

**Pull request merged**